File: API_R_01Agosto2024/Natiele Grazzielly/q1.js
import { question } from "readline-sync"
import { getNumber, print } from "./funcoes.js"

function main(){

    const nome = question('NOME: ')
    const ano = getNumber(`ANO DE NASCIMENTO: `)

    let idade = 2024 - ano  

    //calcular divisores
    let divisores = ''
    let somatorioDivisores = 0
    let qtsDivisores = 0
    let maiorDivisor = idade
    let menorDivisor = 0

    for(let i = 1; i <= idade;i++){
        if(idade % i == 0){
            divisores += " " + i
            somatorioDivisores += i
            qtsDivisores++

        }

        if( qtsDivisores === 2){ // É PRIMO QUANDO SO TEM DOIS(2) DIVISORES, 1 E ELE MESMO , E O MAIOR DIVISOR SEMPRE VAI SER ELE MESMO....
            //SE HOUVER O MAIOR DIVISOR,MOSTRAR N VEZES
            let contador = 1
            while(contador <= idade){
                print(nome)
                contador++
            }
        }else if(qtsDivisores > 2){
            let V_mediaDivisores = somatorioDivisores / qtsDivisores
            print(`NAO HA NUMEROS PRIMOS ENTRE OS DIVISORES DA SUA IDADE,DEVIDO A ISSO VOU MOSTRAR SEU NOME ${V_mediaDivisores} vezes`)
            print(`SOMATORIO DIVISORES: ${somatorioDivisores} --- QTD DIVISORES: ${qtsDivisores}`)
            let contadorDeV = 1
            while(contadorDeV <= V_mediaDivisores){

                print(nome)
                contadorDeV++
            }

        }


    }

    const resultado = 
(`DIVISORES:${divisores}`)

    print(resultado)

}

main()

File: API_R_01Agosto2024/Felipe Gabriel Freitas Marques/Api/q1.py
def main():
    nome = input("Informe seu nome: ")
    ano_de_nascimento = int(input("Informe seu ano de nascimento: "))
    idade = 2024 - ano_de_nascimento
    count = 1
    maior_primo = 0
    count_anterior = 0
    
    print("Os divisores da sua idade são:")
    
    while count <= idade:
        if idade % count == 0:
            print(count)
            if count % count == 0 and count % 3 != 0 and count % 5 != 0 and count % 7 != 0 and count % 2 != 0 :
                if count > count_anterior:
                    maior_primo = count
                count_anterior = count
        count += 1
    
    print(f"O maior divisor primo da sua idade é: {maior_primo}")
    
    count2 = 0
    while count2 < maior_primo:
        print(nome)
        count2 += 1

main()
